clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java

```diff
@@ -284,8 +284,28 @@ public synchronized void seekToEnd(Collection<TopicPartition> partitions) {
             subscriptions.needOffsetReset(tp, OffsetResetStrategy.LATEST);
     }
 
+    // needed for cases where you make a second call to end offsets
+    public synchronized void addEndOffsets(Map<TopicPartition, Long> newOffsets) {
+        innerUpdateEndOffsets(newOffsets, false);
+    }
+
     public synchronized void updateEndOffsets(Map<TopicPartition, Long> newOffsets) {
-        endOffsets.putAll(newOffsets);
+        innerUpdateEndOffsets(newOffsets, true);
+    }
+
+    private void innerUpdateEndOffsets(Map<TopicPartition, Long> newOffsets, boolean replace) {
+        for (Map.Entry<TopicPartition, Long> entry : newOffsets.entrySet()) {
+            List<Long> offsets = endOffsets.get(entry.getKey());
+            if (offsets == null) {
```

Comment:
Should we also create a new list if `replace=true` ?

Reply:
ack